src/env.cc

```diff
@@ -0,0 +1,57 @@
+#include <env.h>
+#include <env-inl.h>
+#include <v8.h>
+
+using v8::HandleScope;
+using v8::Local;
+using v8::Message;
+using v8::StackFrame;
+using v8::StackTrace;
+
+namespace node {
+
+void Environment::PrintSyncWarn() const {
+  if (!warn_on_late_sync_)
+    return;
+
+  HandleScope handle_scope(isolate());
+  Local<v8::StackTrace> stack =
+      StackTrace::CurrentStackTrace(isolate(), 10, StackTrace::kDetailed);
+
+  fprintf(stderr, "WARNING: Detected use of sync API\n");
```

Comment:
I think that's pretty explicit in the fact that the flag now has the word "late".